controller/volume_controller.go

```
@@ -3159,18 +3176,160 @@
+
+			if !e.Status.TargetInstanceReplacementCreated && e.Status.CurrentTargetNodeID != v.Spec.TargetNodeID {
+				log.Debug("Waiting for target instance replacement to be created")
+				return nil
+			}
+
+			if e.Status.CurrentTargetNodeID != v.Spec.TargetNodeID {
+				if e.Status.CurrentState == longhorn.InstanceStateRunning {
+					log.Infof("Suspending engine for v2 data engine live upgrade")
+					e.Spec.DesireState = longhorn.InstanceStateSuspended
+					return nil
+				} else {
```

golangci-lint (1.61.0)

SA9003: empty branch

(staticcheck)